go/test/endtoend/multipooler_manager_test.go

```diff
@@ -1342,3 +1354,590 @@ func TestReplicationAPIs(t *testing.T) {
+
+		// Test write timeout without standby
+		t.Log("Testing write timeout without standby available...")
+		// Create a new connection for this test
+		primaryPoolerClient.Close()
+		primaryPoolerClient, err = NewMultiPoolerTestClient(fmt.Sprintf("localhost:%d", setup.PrimaryMultipooler.GrpcPort))
+		require.NoError(t, err)
+		t.Cleanup(func() { primaryPoolerClient.Close() })
+
+		// This insert should timeout because synchronous_commit=remote_apply requires standby confirmation
+		// Use a 3-second context timeout so the test doesn't wait too long
+		timeoutCtx, timeoutCancel := context.WithTimeout(context.Background(), 3*time.Second)
+		defer timeoutCancel()
+		_, err = primaryPoolerClient.ExecuteQuery(timeoutCtx, "INSERT INTO test_sync_repl (data) VALUES ('test-without-standby')", 0)
+		require.Error(t, err, "Insert should timeout without standby available")
+		assert.Contains(t, err.Error(), "DeadlineExceeded", "Error should indicate a deadline exceeded")
+		t.Log("Write correctly timed out without standby available")
+
+		// Cleanup: Reconnect standby and reset synchronous replication
+		t.Log("Cleanup: Reconnecting standby and resetting synchronous replication...")
+		ctx, cancel = context.WithTimeout(context.Background(), 2*time.Second)
+		defer cancel()
+
+		setPrimaryReq = &multipoolermanagerdata.SetPrimaryConnInfoRequest{
+			Host:                  "localhost",
+			Port:                  int32(setup.PrimaryPgctld.PgPort),
+			StartReplicationAfter: true,
+			StopReplicationBefore: false,
+			CurrentTerm:           1,
+			Force:                 false,
+		}
+		_, err = standbyManagerClient.SetPrimaryConnInfo(ctx, setPrimaryReq)
+		require.NoError(t, err, "SetPrimaryConnInfo should succeed during cleanup")
+
+		// Reset synchronous replication to defaults
+		ctx, cancel = context.WithTimeout(context.Background(), 2*time.Second)
+		defer cancel()
+
+		resetReq := &multipoolermanagerdata.ConfigureSynchronousReplicationRequest{
+			SynchronousCommit: multipoolermanagerdata.SynchronousCommitLevel_SYNCHRONOUS_COMMIT_ON,
+			SynchronousMethod: multipoolermanagerdata.SynchronousMethod_SYNCHRONOUS_METHOD_FIRST,
+			NumSync:           0,
+			StandbyIds:        []*clustermetadatapb.ID{},
+			ReloadConfig:      true,
+		}
+		_, err = primaryManagerClient.ConfigureSynchronousReplication(ctx, resetReq)
+		require.NoError(t, err, "Reset configuration should succeed")
+
+		// Drop test table
+		primaryPoolerClient.Close()
+		primaryPoolerClient, err = NewMultiPoolerTestClient(fmt.Sprintf("localhost:%d", setup.PrimaryMultipooler.GrpcPort))
+		require.NoError(t, err)
+		t.Cleanup(func() { primaryPoolerClient.Close() })
+		_, err = primaryPoolerClient.ExecuteQuery(context.Background(), "DROP TABLE IF EXISTS test_sync_repl", 0)
+		require.NoError(t, err)
+
+		t.Log("End-to-end synchronous replication test completed successfully")
+	})
+
+	t.Run("ConfigureSynchronousReplication_Standby_Fails", func(t *testing.T) {
```

Comment:
Worth adding a test that tries to request ReplicationStatus from a PRIMARY and gets a failed precondition?